apps/dav/lib/Upload/UploadHome.php

```diff
@@ -97,10 +98,43 @@ private function getView() {
 		$rootView = new View();
 		$user = \OC::$server->getUserSession()->getUser();
 		Filesystem::initMountPoints($user->getUID());
-		if (!$rootView->file_exists('/' . $user->getUID() . '/uploads')) {
-			$rootView->mkdir('/' . $user->getUID() . '/uploads');
+
+		$config = \OC::$server->get(IConfig::class);
+
+		$allowSymlinks = $config->getSystemValueBool('localstorage.allowsymlinks', false);
+		$uploadsPath = $config->getSystemValueString('uploads_path', '');
+		$userPath = '/' . $user->getUID() . '/uploads';
+        $absoluteUserPath = $rootView->getLocalFile($userPath);
+
+		if ($allowSymlinks && $uploadsPath !== '' && $absoluteUserPath) {
+			$uploadUserPath = $uploadsPath . $userPath;
+
+			if (!$rootView->file_exists($userPath) || !is_link($absoluteUserPath) || ($uploadUserPath != realpath($absoluteUserPath)) ) {
```

Comment:
We do we check for `!$rootView->file_exists($userPath)`?

Reply:
I would invert the if/else statement to have a more readable condition.

```suggestion
			if ($rootView->file_exists($userPath) && is_link($absoluteUserPath) && ($uploadUserPath === realpath($absoluteUserPath)) ) {
```